src/libsync/owncloudpropagator.cpp

```diff
@@ -389,13 +390,34 @@ void OwncloudPropagator::start(const SyncFileItemVector& items)
                 currentDirJob->append(dir);
             }
             directories.push(qMakePair(item->destination() + "/" , dir));
-        } else if (PropagateItemJob* current = createJob(item)) {
-            if (item->_instruction == CSYNC_INSTRUCTION_TYPE_CHANGE) {
-                // will delete directories, so defer execution
-                directoriesToRemove.prepend(current);
-                removedDirectory = item->_file + "/";
-            } else {
-                directories.top().second->append(current);
+        } else {
+            // Ensure that only new files are inserted into PropagateFiles
+            if (enableBundledRequests && item->_instruction == CSYNC_INSTRUCTION_NEW) {
```

Comment:
It is basis for dispatching bundles in the future, this is why. You can call this feature bundling 1.0. i can change a name however. I also give a flag, because this is now cross-folder, some people might not like it, so I give them the chance to sync folder by folder just by changing one capability on server.